core/src/com/teamturtle/infinityrun/stages/QuizStage.java

```diff
@@ -32,75 +34,100 @@
 
     //    Components
     private TextButton guess1Button, guess2Button, guess3Button;
     private Table parentTable, buttonTable;
+    private List<Word> guesses;
+    private WordLoader wordLoader;
 
     private static final float TEXT_BUTTON_PADDING = 5.0f;
     private static final float PARENT_TABLE_WIDTH = 600.0f, PARENT_TABLE_HEIGHT = 370.0f;
     private static final float PARENT_TABLE_POS_X = 100.0f, PARENT_TABLE_POS_Y = 50.0f;
     private static final float ROW_PADDING = 20.0f;
 
-    public QuizStage(IQuizStageListener handler, List<Word> possibleWords) {
+    public QuizStage(IQuizStageListener handler, List<Word> collectedWords) {
         super(new FitViewport(InfinityRun.WIDTH, InfinityRun.HEIGHT));
         this.handler = handler;
+        this.collectedWords = collectedWords;
 
         skin = new Skin(Gdx.files.internal("skin/uiskin.json"));
         quizLabel = new Label("Quiz-dags!", skin);
+        wordLoader = new WordLoader();
+        wordCategory = collectedWords.get(0).getCategory();
 
-        Random random = new Random();
-
-//        Temporary strings and emoji
-        guess1 = "Banan";
-        guess2 = "Äpple";
-        guess3 = "Päron";
-
-//        TODO: FIX THIS MESS
-        WordRandomizer emojiRandomizer = new WordRandomizer(new WordLoader().getWordsFromCategory(1));
-        Word word = emojiRandomizer.getNext();
-        emoji = new Emoji(word);
-        /*
-        *   In near future guess-strings should be changed to emojis.
-        *   Check if emoji.getName() == chosenGuess.getName()
-        */
-
-        createButtons();
+        guesses = getRandomGuesses(collectedWords);
+        emoji = new Emoji(guesses.get(0));
+        createButtons(guesses);
         createTableUi();
         addActor(parentTable);
         Gdx.input.setInputProcessor(this);
     }
 
-    private void randomizeWords() {
-
+    private List<Word> getRandomGuesses(List<Word> collectedWords) {
+        Random random = new Random();
+        List<Word> returnList = new ArrayList<Word>();
+        while (returnList.size() < 3) {
+            if (collectedWords.size() > 0) {
+                int index = random.nextInt((collectedWords.size() - 1) + 1);
+                if (!returnList.contains(collectedWords.get(index))) {
+                    returnList.add(collectedWords.get(index));
+                }
+                collectedWords.remove(index);
+            }
+            else {
+                List<Word> wordsFromCategory = wordLoader.getWordsFromCategory(wordCategory);
+                int index = random.nextInt((wordsFromCategory.size() - 1) + 1);
+                if (!returnList.contains(wordsFromCategory.get(index))) {
+                    returnList.add(wordsFromCategory.get(index));
+                }
+            }
+        }
+        return returnList;
     }
 
-    private void createButtons() {
-        guess1Button = new TextButton(guess1 + "?", skin);
+    private void createButtons(List<Word> guesses) {
```

Comment:
Det här borde väl fungera?
`for (Word guess : guesses) {`
`int index = random.nextInt((guesses.size() - 1) + 1);`
`TextButton button = new TextButton(guess.getText(), skin);`
`button.addListener(new ChangeListener() {`
`@Override`
`public void changed(ChangeEvent event, Actor actor) {`
`handler.onGuessClick(guess.equals(emoji.getWordModel()));`
`                }`
`            });`
`    button.pad(TEXT_BUTTON_PADDING);`